File: Homework_1/fourargs.cpp
// Copyright 2017 Michael Graziano mjgrazia@bu.edu
#include <iostream>

using namespace std;

int main(int argc, char *argv[]) {

    if (argc == 1) {
        return 0;
    }

    else {

        for(int i = 1; i < argc; i++) {

            if (i <= 4) {
                cout << argv[i] << "\n";
            }
            
            else {
                cerr << argv[i] << "\n";
            }
        }
    }
    
    return 0;
}

File: Homework_1/fourargs.py
# Copyright 2017 Michael Graziano mjgrazia@bu.edu
import sys

if len(sys.argv) >= 2:
    arguments = sys.argv[1::]
    for position in range(len(arguments)):
        # print(position, end = " ")
        if position <= 3:
            # print("Adding value to sys.stdout:")
            sys.stdout.buffer.write((arguments[position]+"\n").encode())
        else:
            # print("Adding value to sys.stderr:")
            sys.stderr.buffer.write((arguments[position]+"\n").encode())

